db/sql_stmt_cache.c

```diff
@@ -112,12 +118,6 @@ static int stmt_cache_requeue_entry(stmt_cache_t *stmt_cache,
         return -1;
     }
 
-    int rc = sqlite3_reset(entry->stmt); // reset vdbe when adding to hash tbl
-    assert(rc == SQLITE_OK);
```

Comment:
Heh, I saw this assert causing a crash in truncatesc test in arm arch and it's strange that we we expected it to always reset successfully. I'm glad you removed the assert.

Reply:
I made `select sleep` catch disconnected client and abort execution. So `sqlite3_reset` was returning `SQLITE3_ABORT` and assert would fail.